### PR TITLE
Remove SolverCompetitionId in favor of AuctionId

### LINKED ISSUE #463: Revert #456 when we switch over to autopilot generated auction ids
The change from https://github.com/cowprotocol/services/pull/456 is not needed after that. We only need it as a temporary measure until then.

### PR DESCRIPTION
**Pull request by vkgnosis**
SolverCompetitionId was a temporary workaround for the lack of a
globally unique auction id. With the autopilot change we have this now
so we can get rid of the old id.

Note that by switching to auction ids from solver competition ids it is technically possible that we would re-use and thus overwrite existing competition info. In practice this will not happen because all the auction ids are already larger than the competition ids. (If it did happen it still wouldn't be a big problem as we didn't want to store competition info forever.)

Closes #463 

### Test Plan

CI

### CODE REVIEW

File: crates/database/src/solver_competition.rs [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
     const QUERY: &str = r#"
-INSERT INTO solver_competitions (json, tx_hash)
-VALUES ($1, $2)
-RETURNING id
-    "#;
-    let (id,) = sqlx::query_as(QUERY)
+INSERT INTO solver_competitions (id, json, tx_hash)
+VALUES ($1, $2, $3)
+ON CONFLICT (id) DO UPDATE
+SET json = EXCLUDED.json, tx_hash = EXCLUDEd.tx_hash
```
Comment (vkgnosis) [inline crates/database/src/solver_competition.rs:14]: This gives us the possibility of overwriting the competition info. Could also leave it out so we get an error if we try to. Maybe error is better because we're not currently intentionally overwriting?

File: crates/model/src/solver_competition.rs [hunk 2]
```diff
@@ -7,11 +8,13 @@ use serde::{Deserialize, Serialize};
...
 
-pub type SolverCompetitionId = i64;
-
 #[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct SolverCompetition {
+    // default annotation because until https://github.com/cowprotocol/services/pull/488 we weren't
+    // storing an auction id in the database.
+    #[serde(default)]
+    pub auction_id: AuctionId,
```
Comment (vkgnosis) [inline crates/model/src/solver_competition.rs:17]: This has the side effect of letting you see the auction id when querying solver competition by tx hash.

File: crates/database/src/solver_competition.rs [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
+) -> Result<(), sqlx::Error> {
     const QUERY: &str = r#"
-INSERT INTO solver_competitions (json, tx_hash)
-VALUES ($1, $2)
-RETURNING id
-    "#;
-    let (id,) = sqlx::query_as(QUERY)
+INSERT INTO solver_competitions (id, json, tx_hash)
+VALUES ($1, $2, $3)
+ON CONFLICT (id) DO UPDATE
```
Comment (nlordell) [inline crates/database/src/solver_competition.rs:13]: Why is this needed now (and wasn't needed before)? Wouldn't this signify that we accidentally reused an auction ID somehow?

**Review by nlordell — APPROVED** [review]
LGTM.

File: crates/database/src/solver_competition.rs (hunk 3 above)
Comment (nlordell) [reply crates/database/src/solver_competition.rs:13]: 🤦 - explanation ☝️

File: crates/database/src/solver_competition.rs (hunk 1 above)
Comment (nlordell) [reply crates/database/src/solver_competition.rs:14]: > Maybe error is better because we're not currently intentionally overwriting?

This would be my inclination. Especially because:

> In practice this will not happen because all the auction ids are already larger than the competition ids.

So if it does, then one of our assumptions is broken.

File: crates/orderbook/openapi.yml [hunk 4]
```diff
@@ -1105,10 +1109,14 @@ components:
         with.
       type: object
       properties:
+        auctionId:
+          type: number
```
Comment (MartinquaXD) [inline crates/orderbook/openapi.yml:1113]: I believe this should be `integer` and to so that it has the same type as the id of the `Auction` type.

File: crates/model/src/auction.rs [hunk 5]
```diff
@@ -1,11 +1,26 @@
 //! Module defining a batch auction.
 
-use crate::{order::Order, solver_competition::SolverCompetitionId, u256_decimal::DecimalU256};
+use crate::{order::Order, u256_decimal::DecimalU256};
 use primitive_types::{H160, U256};
 use serde::{Deserialize, Serialize};
 use serde_with::serde_as;
 use std::collections::BTreeMap;
 
+pub type AuctionId = i64;
```
[the remark this answers is not in the document]
Comment (vkgnosis) [reply crates/model/src/auction.rs:9]: Like Nic said I sometimes prefer seeing a more explicit type than a generic integer. Even stronger would be a newtype `pub struct AuctionId(pub i64)` but that also makes the type less ergonomic so the alias is a good middleground.
